test-locally.sh

```diff
@@ -0,0 +1,93 @@
+#           application-to-test/
+
+# 2. Ensure that child design system projects are looking at local version of the core DS by updating the package.json so that the following dependencies look like this:
+#   "@cmsgov/design-system": "file:../design-system/packages/design-system",
+#   "@cmsgov/design-system-docs": "file:../design-system/packages/design-system-docs",
+#   "@cmsgov/design-system-scripts": "file:../design-system/packages/design-system-scripts",
+
+# 3. Ensure that application project is pointing at local version of child design system by modifying the package.json so that the following dependencies look like this:
+#   "@cmsgov/ds-healthcare-gov": "file:../hcgov-design-system",
+#       or
+#   "@cmsgov/ds-medicare-gov": "file:../mgov-design-system",
+
+# USING THIS SCRIPT
+# In the root directory of the design-system project, run `./test-locally.sh`
+#   This will install dependencies and run the core DS.
+#   Then, it will navigate to the child DS project. It will remove node_modules, install dependencies & build child system.
+#   Then, it will navigate to an application project. It will remove node_modules, install dependencies & build the application.
+
+# This script will try to use the default variables defined in this file, but you can override these for additional testing.
+# For example, if you want to test the hcgov child system, when the default CHILD_DS_NAME variable is mgov child system, you can:
+#   1. Update the CHILD_DS_NAME variable in this file
+#   2. Run `./test-locally.sh -c hcgov-design-system
+
+# Similarly, if you want to change the application path your are testing with, you can:
+#   1. Update the APP_NAME path variable in this file
+#   2. Run `./test-locally.sh -a [directory name of application]
+#     a. If you only want to test in child systems and not in an application, you can run `./test-locally.sh -a ""`
+
+#!/bin/sh
+
+set -e
+source $NVM_DIR/nvm.sh;
+
+RED="\033[0;31m"
+GREEN='\033[0;32m'
+NC='\033[0m' # No color
+
+# either 'mgov-design-system' or 'hcgov-design-system'
+CHILD_DS_NAME="mgov-design-system"
+APP_NAME="coverage-tools-frontend"
+
+while getopts ":c:a:" options; do
+    case "${options}" in
+        c)
+            CHILD_DS_NAME=${OPTARG}
+            ;;
+        a)
+            APP_NAME=${OPTARG}
+            ;;
+  esac
+done
+
+if [ "$CHILD_DS_NAME" = "" ]; then
+      echo "${RED}Error: no path set for child system.
+Make sure you have the CHILD_DS_PATH environment variable set or
+that you have passed a path with the -c option.
+"  
+exit
+fi
+
+echo "${GREEN}Building core design system...${NC}"
+yarn install
+yarn build
+
+echo "${GREEN}Building child design system with local dependencies at path $CHILD_DS_NAME...${NC}"
+cd ../$CHILD_DS_NAME
+rm -rf node_modules
+yarn install
+# yarn build
+
+if [ "$APP_NAME" != "" ]; then
+    echo "${GREEN}Building application with local dependencies at path $APP_NAME...${NC}"
+    cd ../$APP_NAME
+    nvm use
```

Comment:
For applications without a .nvmrc file (eg. `windowshop` & `taxtools`), could we hard code nvm to a valide version eg `nvm use 10.22.1`? Otherwise it'll error out with:
```
error postcss@8.2.6: The engine "node" is incompatible with this module. Expected version "^10 || ^12 || >=14". Got "13.12.0"
error Found incompatible module.
````

Reply:
When I ran the script for coverage tools i ran into an issue 
```
N/A: version "10.16.0 -> N/A" is not yet installed.

You need to run "nvm install 10.16.0" to install it before using it.
```

Wondering if there is a way to prevent this for other applications or if we just add the missing version and move on?

Reply:
That's a good question. I'm wondering if I can do some better error handling for errors... When I briefly googled this the other day, it made it seem like bash is not as robust as Javascript for error handling